### DISCUSSION TITLE
Customize automatic navigation

### DISCUSSION THREAD

**Discussion by e4rthdog** [squidfunk/mkdocs-material#6935] [Q&A] [answered]
We are adding pages constantly on various sections so the sidebar menu updating automatically based on our entries is perfect.

Btu there are some things that i would like to have control for the top and sidebar navigation. For example:

- Be able to influence the order of the pages that are shown
- Be able to name the sidebar items as i wish. It seems that even if i have the `title:` meta on the page, this is not reflected in the sidebar menu. It gets the name from the `.md` filename.

**Comment by alexvoss** [comment]
Have a look at the [awesome pages plugin](https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin) for controlling bits of the navigation while having other bits populated automatically.

Re the `title:` meta element, I am not sure why that would not work, it does for me. Can you create a [minimal reprodution](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) so we can have a look?

**Reply by e4rthdog** [reply]
To be accurate, i want to influence the names of the FOLDERS that appear on the sidebar. I am expecting to create an `index.md` file in the folder and the title attribute to appear on the sidebar menu.

For normal `.md` files in the folder the sidebar shows the title attribute.

**Reply by alexvoss** [reply]
Hm, that does seem like a sensible expectation. I assume you have `navigation.indexes` on, so that might be something to search for in the discussions and issues, to see if this has been covered before.

**Reply by e4rthdog** [reply]
Yes i am using `navigation.indexes`.

In the meantime i got the awesome pages plugin which made me more happy. Thanks, i will also search in the discussions.

**Reply by alexvoss** [reply, marked as answer]
This seems to be related: https://github.com/squidfunk/mkdocs-material/discussions/3321

**Reply by alexvoss** [reply]
The `navigation.indexes` feature is implemented with only a template as far as I can see, `nav-item.html`. Extracting the title from the index page would mean reading from the Markdown file, which is not something you would do in Jinja2. I am pretty sure what you ask for is possible to do but not sure about the tradeoffs and what effort it would take.

**Reply by e4rthdog** [reply]
Yeah, since awesome pages plugin exists then we are ok. The plugin is so good that i would see it being part of the built in for mkdocs-material. Thanks!

**Reply by kamilkrzyskow** [reply]
Similar and same answer from me:
- https://github.com/mkdocs/mkdocs/discussions/3588#discussioncomment-8651798
